draft-ietf-jsonpath-base.md

```diff
@@ -1590,7 +1585,7 @@ the second argument is not a string conforming to {{-iregexp}}.
 | `$[?count(1) == 1]` | Invalid typing since `1` is not a path  |
 | `$[?count(foo(@.*)) == 1]` | Valid typing, where `foo` is a function extension with argument of type `OptionalNodesType` and result type `OptionalNodesType` |
 | `$[?match(@.timezone, 'Europe/.*')]`         | Valid typing |
-| `$[?match(@.timezone, 'Europe/.*') == true]` | Invalid typing since `OptionalBooleanType` is not a `comparable` |
+| `$[?match(@.timezone, 'Europe/.*') == true]` | Invalid typing since `PropositionType` can only appear in `test-expr` |
```

Comment:
I think that this is a worse outcome than the current solution.

Reply:
The current solution is inconsistent and contradictory.  Literally _any_ solution that is consistent is better.